### DISCUSSION TITLE
How to customize slash menu functionality?

### DISCUSSION THREAD

**Discussion by freeseamew** [toeverything/blocksuite#8035] [Q&A] [answered]
I would like to ask how to remove the functions of the slash menu that are not needed (e.g. Todo-list, YouTube embed, etc.).

Also, I would like to ask how to disable the functions that are not needed (e.g. Todo-list, etc.).

**Comment by lawvs** [comment, marked as answer]
Thanks for reaching out. Here is an example of how to customize the `SlashWidget`. You can find more usage in [defaultSlashMenuConfig](https://github.com/toeverything/blocksuite/blob/6d05f9d1286a012514c29d438fbc8fbe4c99648f/packages/blocks/src/root-block/widgets/slash-menu/config.ts#L127-L147).

```ts
import { PageEditorBlockSpecs } from "@blocksuite/blocks";
import { PageEditor } from "@blocksuite/presets";
import {
  AffineFormatBarWidget,
  PageEditorBlockSpecs,
} from "@blocksuite/blocks";

const editor = new PageEditor();
editor.pageSpecs = [...PageEditorBlockSpecs].map((spec) => {
  if (spec.schema.model.flavour === "affine:page") {
    spec = patchPageRootSpec(spec as BlockSpec<"affine:page", PageRootService>);
  }
  return spec;
});

function patchPageRootSpec(spec: BlockSpec<"affine:page", PageRootService>) {
  const setup = spec.setup;
  const newSpec: typeof spec = {
    ...spec,
    setup: (slots, disposable) => {
      setup?.(slots, disposable);
      slots.mounted.once(({ service }) => {
        const pageRootService = service as PageRootService;
        const onWidgetConnected = slots.widgetConnected.on((view) => {
          if (view.component instanceof AffineSlashMenuWidget) {
            setupSlashMenuEntry(view.component);
          }
        });
        disposable.add(onWidgetConnected);
      });
    },
  };

  return newSpec;
}

function setupSlashMenuEntry(slashMenu: AffineSlashMenuWidget) {
  const menu = slashMenu.config.items.slice();
  menu.unshift({
    name: "Custom menu",
    icon: Icon,
    action: ({ rootComponent }) => {
      // Do something
    },
  });

  slashMenu.config = {
    ...AffineSlashMenuWidget.DEFAULT_CONFIG,
    items: menu,
  };
}

```

**Comment by xiaogonggong-w** [comment]
> I would like to ask how to remove the functions of the slash menu that are not needed (e.g. Todo-list, YouTube embed, etc.).我想问一下如何删除斜杠菜单中不需要的功能（例如 Todo-list、YouTube 嵌入等）。
> 
> Also, I would like to ask how to disable the functions that are not needed (e.g. Todo-list, etc.).另外，我想问如何禁用不需要的功能（例如 Todo-list 等）。

How do you do it, and I would be very grateful if you could answer me？

**Reply by raintoway** [reply]
> > I would like to ask how to remove the functions of the slash menu that are not needed (e.g. Todo-list, YouTube embed, etc.).我想问一下如何删除斜杠菜单中不需要的功能（例如 Todo-list、YouTube 嵌入等）。
> > Also, I would like to ask how to disable the functions that are not needed (e.g. Todo-list, etc.).另外，我想问如何禁用不需要的功能（例如 Todo-list 等）。
> 
> How do you do it, and I would be very grateful if you could answer me？

this file `/blocksuite/packages/blocks/src/root-block/widgets/slash-menu/config.ts`, delete item as u wish
